src/diff_walker.rs

```diff
@@ -384,6 +415,33 @@ impl DiffWalker {
         is_split
     }
 
+    fn normalize_const(schema_object: &mut SchemaObject) {
+        fn do_normalize(value: Value) -> SchemaObject {
+            match value {
+                Value::Object(obj) => {
+                    let properties = obj
+                        .into_iter()
+                        .map(|(k, v)| (k, Schema::Object(do_normalize(v))))
+                        .collect::<BTreeMap<_, _>>();
+                    SchemaObject {
+                        object: Some(Box::new(ObjectValidation {
+                            properties,
+                            ..Default::default()
+                        })),
+                        ..Default::default()
+                    }
+                }
+                _ => SchemaObject {
+                    const_value: Some(value),
+                    ..Default::default()
+                },
+            }
+        }
+        if let Some(value) = schema_object.const_value.take() {
+            *schema_object = do_normalize(value)
```

Comment:
can we do `*schema_object.const_value = Some(...)` instead of overwriting the entire schema object? Very unlikely to matter since `const` trumps literally anything else, but I don't think we _have_ to destroy all other properties on `schema_object`

Reply:
what we are doing here is normalizing `const_value` into `object`  with each properties are constant value (las to res in https://github.com/getsentry/json-schema-diff/issues/15#issuecomment-1544047246). so we could write `*schema_object.object = Some(....)`, instead of overwriting `const_value`

Reply:
whoops nvm. right.